src/Microsoft.AspNetCore.Mvc.Core/Formatters/StringOutputFormatter.cs

```diff
@@ -29,18 +29,17 @@ public override bool CanWriteResult(OutputFormatterCanWriteContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            // Ignore the passed in content type, if the object is string
-            // always return it as a text/plain format.
             if (context.ObjectType == typeof(string) || context.Object is string)
             {
-                if (!context.ContentType.HasValue)
+                // Call into base to check if the current request's content type is a supported media type.
+                var canWriteResult = base.CanWriteResult(context);
```

Comment:
I actually don't like to use this big hammer method here...if there were a IsSupportedMediaType method exposed by the base, then it would be pretty clear in my opinion...